### DISCUSSION TITLE
그룹#1 회고

### DISCUSSION THREAD

**Discussion by godrm** [codesquad-members-2022/swift-pokergame#81] [Show and tell]
여기에 각자 이번주 회고를 기록해주세요.
- 학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)
- 학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)
- 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)
- 학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!

**Comment by TaeKyeongKim** [comment]
## [학습하면서 잘했던 점, 좋았던 점] 
 - 직접 스토리보드를 사용하면서 시뮬레이터에  화면을 처음으로 만들수 있었던게 좋았습니다. 
 - Struct 와 Class 의 차이점을 조금더 자세하게 이해 했다. 
 - OOP 설계를 할때 모듈별로 나누어서 구현하라는 의미를 생각으로는 알고 있었는데 예제를 보면서 조금더 실질적으로 이해할수 있었다. 
 
## [학습하면서 스스로 아쉬웠던 점, 부족했던 점]
 - 생각을 실행으로 옮기는 시간이 너무 길다. -> 머리로 한번 생각하고 될것 같으면 직접 코딩을 해보면서 시도를 하면되는데, 안될것같은 이유을 계속 머리속으로 생각하고 실행으로 옮기지 않는다. 
 - 이로인해 구체적인 문제가 주어지지 않았을때 혼자서 추상적인 문제의 단계를 구체적으로 나누고 구현해보는 데 까지의 시간이 너무 많이 든다. 
   
## [새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼것]
- 주어진 문제를 모듈화 해서 작게 나누는 연습을 많이 해서 설계 하는 시간을 줄여야겠다. 
- 입, 출력 모듈에서 많이 사용될것같은 함수들을 최대한 추상화 시켜 다른 프로젝트에서도 사용할수 있게 구현해봐야겠다.

**Comment by SangHwi-Back** [comment]
👍 이번 주에는 이런 걸 잘했습니다.

* 이번 주는 Git 오류를 확인하지 못했습니다. Sourcetree와 CLI의 콜라보가 아주 훌륭했습니다.
* 객체 지향이란 거 감이 좀 오는 것 같습니다. Nested Type이니 Generic이니 하는 것들을 이용해서 공통 유틸리티도 만들어보고 코드를 아주 화려하게 장식하고 있어서 기분이 좋습니다.
* 운동을 꾸준히 하고 있습니다.   
예전부터 먹던 유청 단백질을 다 먹어가는데 근손실 방지를 위해 빨리 먹고 한 포 더 구매할겁니다.

👎 이번 주에는 이런 걸 못했습니다.

* 2주차 과제까지 다 진행하지 못했습니다. 반도 진행하지 못한 건 좀 더 노력해야 되는 부분인 것 같습니다.
* 코딩테스트..... 언제나 부족합니다. 항상 더 많이 공부하고 풀어봐야 합니다.

😭 다음 주에는 이런 걸 해볼 겁니다.

* 제가 직접 라이브러리를 만들어 볼 수 있는 것 같습니다. 이런 걸 만들어 놓으면 복잡한 뷰를 만들어 놓고 나중에도 쓸 수 있을 것 같은데 한번 실습해보고 싶네요.
* 동시성 프로그래밍에 대한 블로그 포스팅을 준비하던게 있는데 멈춰 있습니다. 빨리빨리 진행해야겠습니다!

## 여러분들 이번주도 고생 많으셨습니다!

**Comment by rising-jun** [comment]
#### 학습하면서 잘했던 점, 좋았던 점은 무엇인가요?
- git에 대해 조금씩 익숙해지고 있습니다! 저번주에 비해 훨씬 깃에 쓰는 시간이 줄어들었습니다.
- oop 설계에 대해, 변수명 짓는거에 대해 조금씩 나아지고 있다 생각되어 좋았습니다!
#### 학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 
- 6시 이후 미션은 미뤄두고 많이 쉬었던것 같습니다.. 이번주 쉬었으니 다음주는 더 열심히 하겠습니다!
#### 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? 
- step5 화면에 카드 추가를 시도해볼 것 같습니다!

**Comment by JasonLee0223** [comment]
## 학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)
- Git 공부를 하여 이제 PR 보내는 부분이 많이 익숙해지고 오류없이 진행할 수 있었습니다.   
  CLI환경과 GUI환경을 적절히 사용하는게 얼마나 좋은지 느낄 수 있었습니다.
- 목표(To do list)를 작성할 때 세분화해서 만들어보니 성취감도 많이 느끼고 좀 더 세분화된 계획으로 이전보다 많이 학습방법이 개선되었습니다.
- 지난주까지 몸에 안잡혀있던 바이오리듬이 잡혔습니다.

## 학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)
- 신체적인 부분에서 조금이라도 신경쓰이는 부분이 생기면 집중력이 급격하게 떨어져 잠자리나 습관들을 조심해야겠습니다.
- 이번주는 행사가 많아서 밖에 나가는 일이 많았는데 확실히 집중력이 좀 떨어져 주의해야겠습니다.

## 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)
- 추가학습거리를 많이 진행하지 못했던 부분이 아쉬웠습니다.
- 포커게임 Step4까지 잘 진행해봐야겠습니다!

## 학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!
- 여러분 모두 감사합니다!!

**Comment by Damagucci-Juice** [comment]
![image](https://user-images.githubusercontent.com/50472122/155680151-688e5a2a-51f4-46ba-b266-c66c2bf54252.png)

❤️ 잘한 것
* 타임 스케쥴 작성
* 딥워크를 읽었다. 
    * 느슨해진 공부에 긴장감을 주는 책이였다..
* 점심에 산책 루틴 추가 - 광합성에 그만이다. 돌아오는 길에 커피까지 테이크 아웃하면 나도 뉴요커 😎
* 오프라인 모임에 참석한 것
    * 공부는 못했지만 온라인에서 보던 멤버를 실제로 보니 너무 반가웠다. 
* 주에 2번 있는 JK 세미나 모두 노트 필기하며 안졸았다
    * 블로그에 필기본을 올리니 작성할 것도 생기고 좋았다. 
      
🥲 못한 것
* 미션을  스텝 3 까지 밖에 못했다. 스케쥴은 잘 잡혔는데 코드량은 줄어들었다.
    * 오후 시간에 생각보다 느슨했다.
* CS 스터디에서 공부를 못해갔다. -> `아무것도 하기 싫어증` 에 걸린것을 핑계로 안했는데, 왜 안했는지 설명하는게 더 힘들었다. 그리고 약속 시간이 다가올수록 안한 거에 대한 압박감 때문에 편히 쉬지도 못했다. 그냥 적은양이라도 조사를 하고 발표하자. 그게 예의에 맞지 않나 싶다.

🚀 할 것
* 블로깅을 하자. 공부한 것을 기록해야 그나마 조금이라도 남는 것 같다.
* 의심하지 않기 : 발전하고 있는가 나아지고 있는가에 대한 확신이 부족한데 그럴 시간에 코드 하나 더짜기
* CS 스터디 빼먹지 않기

👍 칭찬해요!
* 에보니 : 드립쳐도 받아줘서 칭찬해요!
* 주방 : 항상 리액션 해줘서 칭찬해요!
* 메이스(a.k.a 메시아) : 좋은 강의...  칭찬해요!
* 벡 : 수염이 멋져서 칭찬해요!
* 제이슨 : iMac Mini를 사서 칭찬해요!
* 샐리 : 아프지 마세요 ㅠㅠ
* 카이 : 새벽에 빡코딩 칭찬해요!

**Comment by sanghyeok-kim** [comment]
### 좋았던 점
- 메소드 네이밍의 중요성, 객체간 역할 분리같은 객체지향 프로그래밍의 요소들을 배울 수 있어서 좋았습니다.
- Upstream을 만들어서 내 저장소와 fork한 원본 저장소간 동기화하는 것에 대한 개념이 헷갈렸는데 결국 삽질 끝에 터득했습니다.

### 아쉬웠던 점
- 미션을 다 끝마치지 못한게 제일 아쉽습니다...
- 저녁식사 이후의 시간을 제대로 활용하지 못한 것 같습니다.

### 시도해볼 점
- 주말동안 남은 미션을 시도해 보고, 밀린 Readme 작성도 할 예정입니다.

**Comment by LeeJoobang** [comment]
학습하면서 잘했던 점, 좋았던 점은 무엇인가요? 
 - 객체지향에 대해 생각해볼 수 있었습니다.
 
학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)
- 학습한 내용에 대해 반드시 기록으로 남기고 이것을 온전히 정리하지 못했습니다.
- 정리의 우선순위(학습정리 > TIL > README)대로 작성하기

새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)
- Step2를 진행 중에 있는데 Step4까지 해보는 것입니다.

학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!
- 팀원 모두 함께 할 수 있어 즐거웠습니다.
- 싱하, 데일, 메이스 모르는 개념에 대해 배울 수 있어 무척 좋았습니다.
